datalad/interface/base.py

```diff
@@ -304,3 +308,53 @@ def call_from_parser(cls, args):
         except KeyboardInterrupt as exc:
             ui.error("\nInterrupted by user while doing magic: %s" % exc_str(exc))
             sys.exit(1)
+
+    @staticmethod
+    def _prep(
+            path=None,
+            dataset=None,
+            recursive=False,
+            recursion_limit=None,
+            mark_recursive=False):
+        from .utils import get_normalized_path_arguments
+        from .utils import get_paths_by_dataset
+
+        # upfront check prior any resolution attempt to avoid disaster
+        if path is None and dataset is None:
+            raise InsufficientArgumentsError(
+                "at least a dataset or a path must be given")
+
+        path, dataset_path = get_normalized_path_arguments(
+            path, dataset, default=curdir)
+        content_by_ds, unavailable_paths, nondataset_paths = \
+            get_paths_by_dataset(path,
+                                 recursive=recursive,
+                                 recursion_limit=recursion_limit,
+                                 mark_recursive=mark_recursive)
+        if dataset_path and not content_by_ds and not unavailable_paths:
+            # we got a dataset, but there is nothing actually installed
+            nondataset_paths.append(dataset_path)
+        # complain about nondataset and non-existing paths
+        if nondataset_paths:
+            raise ValueError(
+                "will not touch paths outside of installed datasets: %s"
+                % nondataset_paths)
+        return content_by_ds, unavailable_paths
+
+
+def report_result_objects(cls, res, args, passive):
+    from datalad.ui import ui
+    from datalad.distribution.dataset import Dataset
+    if not res:
+        ui.message("Nothing was {}".format(cls._passive))
+        return
+    msg = "{n} {obj} {action}:\n".format(
+        obj='items were' if len(res) > 1 else 'item was',
+        n=len(res),
+        action=cls._passive)
+    for item in res:
+        if isinstance(item, Dataset):
+            msg += "Dataset: %s\n" % item.path
+        else:
+            msg += "File: %s\n" % item
```

Comment:
How to extend for other types of objects and reports? Eg in case of crawl and ls?